modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/OvernightInArrearsCapletFloorletPeriodAmounts.java

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (C) 2024 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.capfloor;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.joda.beans.Bean;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableMap;
+import com.opengamma.strata.product.capfloor.OvernightInArrearsCapletFloorletPeriod;
+
+/**
+ * A map of double values keyed by overnight rate in arrears caplet/floorlet periods.
+ * <p>
+ * The double values should represent the same metric.
+ */
+@BeanDefinition
+public final class OvernightInArrearsCapletFloorletPeriodAmounts
+    implements ImmutableBean, Serializable {
+
+  /**
+   * The map of overnight rate in arrears caplet/floorlet periods to the double amount.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableMap<OvernightInArrearsCapletFloorletPeriod, Double> amounts;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance of double amounts.
+   *
+   * @param doubleMap the map of doubles
+   * @return the instance
+   */
+  public static OvernightInArrearsCapletFloorletPeriodAmounts of(Map<OvernightInArrearsCapletFloorletPeriod, Double> doubleMap) {
+    return new OvernightInArrearsCapletFloorletPeriodAmounts(doubleMap);
+  }
+
+  /**
+   * Gets a double amount for the provided overnight rate in arrears caplet/floorlet.
+   *
+   * @param period the caplet/floorlet
+   * @return the double amount, empty if missing
+   */
+  public Optional<Double> findAmount(OvernightInArrearsCapletFloorletPeriod period) {
+    return Optional.ofNullable(amounts.get(period));
+  }
+
+  /**
+   * Gets a double amount for the provided overnight rate in arrears caplet/floorlet.
+   *
+   * @param period the caplet/floorlet
+   * @return the double amount
+   * @throws IllegalArgumentException if the period is missing
+   */
+  public double getAmount(OvernightInArrearsCapletFloorletPeriod period) {
+    if (!amounts.containsKey(period)) {
+      throw new IllegalArgumentException("Could not find double amount for " + period);
+    }
+    return amounts.get(period);
```

Comment:
I would use the `findAmount()` above, and then you can return using `.orElseThrow()`, i.e.:

```java
findAmount(period).orElseThrow(() -> new IllegalArgumentException("Could not find double amount for " + period));
```